lib/game_client/ogs/ogs_game.dart

```diff
@@ -220,6 +266,10 @@ class OGSGame extends Game {
 
       // Update game result if the game has ended
       final phase = gameData['phase'] as String?;
+      if (phase != null && phase != _gamePhase) {
+        _logger.info('Game $id phase updated: $_gamePhase -> $phase');
+        _gamePhase = phase;
```

Comment:
phase comes in from a couple places.  We get it from "gamedata" because that's the only place to get it at the start of a game.  We also get it from "phase" because that fires during the "play" -> "stone removal" transition.  Some events, like game finished, will fire both messages.